File: sample-java-app/src/main/java/com/unissey/samplejavaapp/ui/SharedViewModel.java
package com.unissey.samplejavaapp.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.SavedStateHandle;
import androidx.lifecycle.ViewModel;

import com.unissey.sdk.core.domain.UnisseySession;

public class SharedViewModel extends ViewModel {

    private final SavedStateHandle state;

    public LiveData<String> getVideoUri() {
        return state.getLiveData("videoUri");
    }

    public void setVideoUri(String videoUri) {
        state.set("videoUri", videoUri);
    }

    public UnisseySession unisseySession = null;
    public boolean isCameraStarted = false;
    public boolean isVideoCaptureRunning = false;
    public MutableLiveData<String> instructionText = new MutableLiveData<>();

    public SharedViewModel(SavedStateHandle savedStateHandle) {
        state = savedStateHandle;
    }
}

File: sample-legacy-app/src/main/java/com/unissey/samplelegacyapp/ui/SharedViewModel.kt
package com.unissey.samplelegacyapp.ui

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import com.unissey.sdk.core.domain.UnisseySession

class SharedViewModel(private val savedStateHandle: SavedStateHandle) : ViewModel() {

    val videoUri = savedStateHandle.getLiveData<String>("videoUri")
    var isCameraStarted = false
    var isVideoCaptureRunning = false
    var instructionText = MutableLiveData<String>()
    var unisseySession: UnisseySession? = null

    fun setVideoUri(videoUri: String) {
        savedStateHandle["videoUri"] = videoUri
    }
}